physionet-django/events/models.py

```diff
@@ -153,3 +153,49 @@ def withdraw(self, comment_to_applicant=None):
             comment_to_applicant (str): The comment to add to the participant about the status.
         """
         self._apply_decision(self.EventApplicationStatus.WITHDRAWN, comment_to_applicant)
+
+
+class EventDataset(models.Model):
+    """
+    Captures information about datasets for events.
+    """
+    class EventDatasetStatus(models.TextChoices):
+        Active = 'AC', _('Active')
+        Inactive = 'IN', _('Inactive')
+
+    class EventDatasetAccessType(models.TextChoices):
+        GOOGLE_BIG_QUERY = 'GBQ', _('Google BigQuery')
+        RESEARCH_ENVIRONMENT = 'RE', _('Research Environment')
+
+    event = models.ForeignKey(Event, on_delete=models.CASCADE, related_name='datasets')
+    dataset = models.ForeignKey("project.PublishedProject", on_delete=models.CASCADE)
+    access_type = models.CharField(default=EventDatasetAccessType.GOOGLE_BIG_QUERY, max_length=10,
+                                   choices=EventDatasetAccessType.choices)
+    status = models.CharField(default=EventDatasetStatus.Active, max_length=2,
+                              choices=EventDatasetStatus.choices)
+    added_datetime = models.DateTimeField(auto_now_add=True)
+    updated_datetime = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return self.event.title + ' -- ' + self.dataset.title
+
+    def is_accessible(self):
+        """
+        checks if the dataset is accessible to the user
+        Dataset is accessible if:
+        1. The dataset is active
+        2. The event has not ended
+        """
+        if self.status == self.EventDatasetStatus.Inactive:
+            return False
+
+        if self.event.end_date < timezone.now().date():
```

Comment:
Good catch. i think this issue might be in Event as well. i will check and open a separate PR for that.

I made the suggested change in this PR also

Reply:
I think the current code implementation is correct.

Our goal is to let users access dataset till the last day

Case 1
Say last day of event is 25 Feb. 2023, and the current date is also  25 Feb. 2023,
This means the if statement will be False. and the function will return True. hence the data will be accessible.

Case 2
Say last day  of event is 25 Feb. 2023, and the current date is 26 Feb. 2023,
This means the if statement will be True. and the function will return False. hence the data will not be accessible.

![image](https://user-images.githubusercontent.com/24412619/220395129-b5c93514-e975-4e3f-8d04-7219f3b40e2d.png)


![image](https://user-images.githubusercontent.com/24412619/220394712-026d480c-e7ab-4b49-93e6-b5baaa06362a.png)

Reply:
Because, the function works like this
Check all the cases where we might reject the access. if we can reject access for any reason, reject it rightaway
if not keep checking, and finally after everything is checked, and we have no reason to reject access, Grant the access.

Reply:
I might be confused here. So let me do a quick test on the local setup

Reply:
Resolved after discussing with @alistairewj  on slack.

We decided to improve the readibiliy by using
`if now_date > end_date -> return False`